containers/docker/plugin_list.go

```diff
@@ -13,18 +14,21 @@ import (
 // PluginList lists all plugins on a target. If instance is not empty, it will return a plugin
 // named `instance` if it exists.
 func (m *Manager) PluginList(ctx context.Context, instance string) (*cpb.ListPluginsResponse, error) {
-	var args filters.Args
-	if instance != "" {
-		args = filters.NewArgs(filters.KeyValuePair{Key: "name", Value: instance})
-	}
-
-	resp, err := m.client.PluginList(ctx, args)
+	resp, err := m.client.PluginList(ctx, filters.Args{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to list plugins: %w", err)
 	}
 
 	res := &cpb.ListPluginsResponse{}
 	for _, plugin := range resp {
+		if instance != "" {
+			// plugin.Name will have format <name>:<tag>.
+			// instance_name (from StartPluginRequest) does not have a tag,
+			// so cut off only the name here.
+			if pluginName, _, _ := strings.Cut(plugin.Name, ":"); pluginName != instance {
+				continue
+			}
+		}
```

Comment:
Could we have a test that cover this?

Reply:
The "multiple-plugins" testcase covers this, in the latest patch i removed the code that was mocking handling of the `name` arg, since it was not reflecting the actual docker behaviour.